bundle/src/main/java/com/adobe/acs/commons/ccvar/impl/AllPagePropertiesContentVariableProvider.java

```diff
@@ -0,0 +1,100 @@
+package com.adobe.acs.commons.ccvar.impl;
+
+import com.adobe.acs.commons.ccvar.ContentVariableProvider;
+import com.adobe.acs.commons.ccvar.PropertyConfigService;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.PageManager;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ValueMap;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+import static com.adobe.acs.commons.ccvar.util.PropertyAggregatorUtil.addPropertiesToMap;
+
+@Component(service = ContentVariableProvider.class)
+public class AllPagePropertiesContentVariableProvider implements ContentVariableProvider {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AllPagePropertiesContentVariableProvider.class);
+
+    public static final String PAGE_PROP_PREFIX = "page_properties";
+    private static final String INHERITED_PAGE_PROP_PREFIX = "inherited_page_properties";
+
+    @Reference
+    private PropertyConfigService propertyConfigService;
+
+    @Override
+    public void addProperties(Map<String, Object> map, SlingHttpServletRequest request) {
+        Resource resource = request.getResource();
+
+        PageManager pageManager = resource.getResourceResolver().adaptTo(PageManager.class);
+        if (pageManager == null) {
+            LOG.warn("PageManager was null, skipping properties.");
+            return;
+        }
+        Page page = pageManager.getContainingPage(resource);
+
+        if (page == null) {
+            LOG.warn("No containing page found for resource at {}", resource.getPath());
+            return;
+        }
+
+        // Add current page properties
+        addPagePropertiesToMap(map, page, PAGE_PROP_PREFIX, propertyConfigService);
+
+        // Add inherited page properties
+        while (page != null) {
+            addPagePropertiesToMap(map, page, INHERITED_PAGE_PROP_PREFIX, propertyConfigService);
+            page = page.getParent();
+        }
+    }
+
+    /**
+     * Add the properties of a page to the given map.  Excluded properties are found in the
+     * {@link PropertyConfigService} service.
+     *
+     * @param map                   the map that should be updated with the properties and their values
+     * @param page                  the page containing properties
+     * @param prefix                the prefix to apply to the
+     * @param propertyConfigService the {@link PropertyConfigService} used to check type and exclusion
+     */
+    private void addPagePropertiesToMap(Map<String, Object> map, Page page, String prefix,
+                                              PropertyConfigService propertyConfigService) {
+        ValueMap pageProperties = page.getProperties();
+        addPropertiesToMap(map, pageProperties.entrySet(), prefix, false, propertyConfigService);
```

Comment:
nit: I was confused as heck trying to find this function until I pulled the code up in an IDE and noticed it was in italics, meaning you statically imported it.  I feel it would be more readable to call explicitly `PropertyAggregatorUtil.addPropertiesToMap()` but again that's maybe just a personal preference.

Reply:
i agree - in other places i'm referencing the utils explicitly so i've updated this